### ISSUE TITLE
Increase type checking before computation

### ISSUE THREAD

**Issue by gnekt** [nunziati/verify-catastrophic-forgetting#1]
The constructor takes padding as a default string parameter, but if someone tries to instantiate a new object with padding initialized to something that is not a:

- int
- str
- tuple

An unexpected exception will raise, cause torch.nn.Conv2d accept one of this type. 

> padding ([int](https://docs.python.org/3/library/functions.html#int), [tuple](https://docs.python.org/3/library/stdtypes.html#tuple) or [str](https://docs.python.org/3/library/stdtypes.html#str), optional) 
> – Padding added to all four sides of the input. Default: 0

Code refer
If <if-statement> will be false, super-constructor is called without control.

https://github.com/nunziati/verify-catastrophic-forgetting/blob/f22c97b2c64ee5e2791da14ecf3e1bfd9eb7fc26/nets.py#L15
